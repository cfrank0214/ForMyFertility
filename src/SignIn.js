
import React, { Component } from 'react';
import {
  Container,
  Body,
  Button,
  Text,
  Header,
  Content,
  Form,
  Icon,
  Title,
  Left,
  Right,
  Item,
  Input,
  Label
} from 'native-base';
import { Auth } from 'aws-amplify'

export default class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: false,
      email: "",
      password: "",
    };
  }

  validateForm() {
    return this.state.email.length > 0 && this.state.password.length > 0;
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  handleSubmit = async event => {
    event.preventDefault();
    this.setState({ isLoading: true })
    try {
      await Auth.signIn(this.state.email, this.state.password);
      console.log('successful sign in!')
      this.props.screenProps.authenticate(true)
    } catch (e) {
      alert(e.message);
      this.setState({ isLoading: false });
    }
  }

  render() {
    return (
      <Container>
      <Header>
          <Left>
            <Button
              transparent
              onPress={() => this.props.navigation.toggleDrawer()}>
              <Icon name="menu" />
            </Button>
          </Left>
          <Body>
            <Title>Sign In</Title>
          </Body>
          <Right />
        </Header>
        <Content padder>
          <Form>
            <Item stackedLabel>
              <Label>Email</Label>
              <Input
                autoFocus = {true}
                keyboardType="email-address"
                autoCorrect={false}
                autoCapitalize="none"
                onChangeText={(email) => { this.setState({ email }); }}
                value={this.state.email}
              />
            </Item>
            <Item stackedLabel last>
              <Label>Password</Label>
              <Input
                secureTextEntry
                onChangeText={(password) => { this.setState({ password }); }}
                value={this.state.password}
              />
            </Item>
          </Form>
          <Button
            block
            success
            disabled={!this.validateForm()}
            isLoading={this.state.isLoading}
            onPress={this.handleSubmit}
          >
            <Text>
              Sign In
              </Text>
          </Button>
        </Content>
      </Container>
    );
  }
}