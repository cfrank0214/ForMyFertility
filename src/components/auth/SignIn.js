import React, { Component } from 'react';
import {
  Container,
  Body,
  Button,
  Form,
  Text,
  Header,
  Content,
  Item,
  Input,
  Label
} from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';
import { Auth, Logger, JS } from 'aws-amplify';

const logger = new Logger('SignIn');

export default class SignIn extends Component {
  constructor(props) {
    super(props);
    this.signIn = this.signIn.bind(this);
    this.checkContact = this.checkContact.bind(this);
    this.changeState = this.changeState.bind(this);
    this.inputs = {};
    this.state = {
      error: null,
      username: null,
      password: null,
    }
  }

  changeState(state, data) {
    const { onStateChange } = this.props;
    if (onStateChange) {
      onStateChange(state, data);
    }
  }

  signIn() {
    const { username, password } = this.state;
    logger.info('sign in with ' + username);
    Auth.signIn(username, password)
      .then(user => this.signInSuccess(user))
      .catch(err => this.signInError(err));
  }

  signInSuccess(user) {
    logger.info('sign in success', user);
    this.setState({ error: null });
    // There are other sign in challenges we don't cover here.
    // SMS_MFA, SOFTWARE_TOKEN_MFA, NEW_PASSWORD_REQUIRED, MFA_SETUP ...
    if (user.challengeName === 'SMS_MFA' || user.challengeName === 'SOFTWARE_TOKEN_MFA') {
      this.changeState('confirmSignIn', user);
    } else {
      this.checkContact(user);
    }
  }

  signInError(err) {
    logger.info('sign in error', err);
    /*
      err can be in different structure:
        1) plain text message;
        2) object { code: ..., message: ..., name: ... }
    */
    this.setState({ error: err.message || err });
  }

  checkContact(user) {
    Auth.verifiedContact(user)
      .then(data => {
        if (!JS.isEmpty(data.verified)) {
          this.changeState('signedIn', user);
        } else {
          user = Object.assign(user, data);
          this.changeState('verifyContact', user);
        }
      });
  }

  render() {
    const { authState, authData } = this.props;
    if (!['signIn', 'signedOut', 'signedUp'].includes(authState)) { return null; }
    const { error } = this.state;
    return (
      <Content >
        <Grid>
          <Form  >
            <Item stackedLabel style={{ marginTop: 30 }}>
              <Label>Email (Username)</Label>
              <Input
                type="text"
                autoCapitalize="none"
                onChangeText={(username) => this.setState({ username })}
                value={this.state.username}
                autoFocus
              />
            </Item>
            <Item stackedLabel last>
              <Label>Password</Label>
              <Input
                type="password"
                secureTextEntry
                onChangeText={(password) => this.setState({ password })}
                value={this.state.password}
              />
            </Item>
            <Button block success onPress={() => this.signIn()}>
              <Text>Sign In</Text>
            </Button>
            {error &&
              <Text>{error}</Text>
            }
            <Row>
              <Button transparent onPress={() => this.changeState('signUp')}>
                <Text> Sign Up</Text>
              </Button>
              <Button transparent onPress={() => this.changeState('forgotPassword')} >
                <Text>Forgot Password</Text>
              </Button>
            </Row>
          </Form>
        </Grid>
      </Content>
    )
  }
}