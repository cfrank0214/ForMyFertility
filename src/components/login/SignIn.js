
import React, { Component } from 'react';
import {
  Container,
  Body,
  Button,
  Text,
  Header,
  Content,
  Form,
  Item,
  Input,
  Label
} from 'native-base';
import { Auth } from 'aws-amplify'

export default class SignIn extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: false,
      email: "",
      password: "",
    };
  }

  validateForm() {
    return this.state.email.length > 0 && this.state.password.length > 0;
  }

  handleChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  handleSubmit = async event => {
    event.preventDefault();
    this.setState({ isLoading: true })
    try {
      await Auth.signIn(this.state.email, this.state.password);
      console.log('successful sign in!')

      this.props.navigation.navigate('Questions');
    } catch (e) {
      alert(e.message);
      this.setState({ isLoading: false });
    }
  }

  render() {
    return (
      <Container>
        <Header />
        <Content padder >
          <Form>
            <Body style={{ marginTop: 10 }}>
              <Text>
                We take securing and protecting
                your information very seriously.
                Which is why we're asking you to
                create a free account before we
                ask you private questions about your
                body and fertility history.
            </Text>
            </Body>
            <Item stackedLabel style={{ marginTop: 30 }}>
              <Label>Email</Label>
              <Input
                autoFocus={true}
                keyboardType="email-address"
                autoCorrect={false}
                autoCapitalize="none"
                onChangeText={(email) => { this.setState({ email }); }}
                value={this.state.email}
              />
            </Item>
            <Item stackedLabel last>
              <Label>Password</Label>
              <Input
                secureTextEntry
                onChangeText={(password) => { this.setState({ password }); }}
                value={this.state.password}
              />
            </Item>
          </Form>
          <Button
            block
            success
            disabled={!this.validateForm()}
            isLoading={this.state.isLoading}
            onPress={this.handleSubmit}
          >
            <Text>
              Sign In
              </Text>
          </Button>
        </Content>
      </Container>
    );
  }
}