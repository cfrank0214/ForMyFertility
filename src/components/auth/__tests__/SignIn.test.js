/**
 * @jest-environment jsdom
 */
import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SignIn from '../SignIn';
import {
  Button,
  Form,
  Text,
  Header,
  Content,
  Item,
  Input,
  Label
} from 'native-base';
import { Col, Row, Grid } from 'react-native-easy-grid';

describe('>>>SignIn --- Shallow Render REACT COMPONENTS', () => {
  let wrapper

  beforeEach(() => {

    const props = {
      inputs: jest.fn(),
      state: jest.fn(),
      authState: 'signIn'
    }

    wrapper = shallow(<SignIn {...props} />)

  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <SignIn />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1)
  });
  it('The <Button /> component exists', () => {
    expect(wrapper.find(Button).exists()).toBe(true);
  });
  it('contains 3 <Button /> components', () => {
    expect(wrapper.find(Button).length).toBe(3);
  });
  it('contains 1 <Form /> component', () => {
    expect(wrapper.find(Form).length).toBe(1);
  });
  it('contains 2 <Input /> component', () => {
    expect(wrapper.find(Input).length).toBe(2);
  });
  it('contains 1 <Row /> component', () => {
    expect(wrapper.find(Row).length).toBe(1);
  });
});