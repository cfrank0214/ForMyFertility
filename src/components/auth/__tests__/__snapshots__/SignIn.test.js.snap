// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>SignIn --- Shallow Render REACT COMPONENTS renders correctly 1`] = `null`;
